bep032tools/generator/tests/test_BEP032Generator.py

```diff
@@ -184,7 +185,42 @@ def test_generate_example_structure(self):
     def doCleanups(self):
         initialize_test_directory(clean=True)
 
+class Test_FolderGeneration(unittest.TestCase):
+
+    def setUp(self):
+        self.test_dir = Path(initialize_test_directory(clean=True))
+        os.mkdir(self.test_dir / 'generateTest')
+        csv_filename = generate_simple_csv_file()
+        self.csv_file = csv_filename
+
+    def test_generate_folder(self):
+        """
+        Test if the Path exists. After generation, check that the folder exists and display the list of sub-folders.
```

Comment:
Hi @killianrochet, i think the test your are aiming to implement should be more specific. You can read again the issue you raised here (#123) and try to cover the two cases your were describing in that issue (potentially in 2 separate `test_` methods.
Within such a test method you should 1) try to generate the situation in which the error was observed 2) run the generator and 3) assert that the observed error does not occur any more.

So a practical next step would be to split this test in 2 methods `test_no_duplicate_folder_generated` and `test_generate_folder` (to ensure the error you reported  if the folder does not yet exist does not occur).

Reply:
Ok, I have separated the different tests into two different test functions